crates/services/producer/src/block_producer/tests.rs

```diff
@@ -543,6 +543,86 @@ mod produce_and_execute_block_txpool {
     }
 }
 
+// Tests for the `dry_run` method.
+mod dry_run {
+    use super::*;
+
+    #[tokio::test]
+    async fn dry_run__executes_with_given_timestamp() {
```

Comment:
What if time is earlier than the last block? Is that valid?

Reply:
Good question, I'd assume that it's valid because it would allow for testing contrafactuals. I.e. "what if this transaction had been submitted in the past, would it have been valid?". Though we do error on previous block heights (which we shouldn't do though https://github.com/FuelLabs/fuel-core/issues/2062).

Reply:
I'll add a test case for it.

Reply:
Added in https://github.com/FuelLabs/fuel-core/pull/2206/commits/c8f3ee1b540bf35a0cf2fc3cd8ed9fa48bb9223e and it works.